src/main/java/edu/harvard/iq/dataverse/license/TermsOfUseForm.java

```diff
@@ -0,0 +1,57 @@
+package edu.harvard.iq.dataverse.license;
+
+import java.io.Serializable;
+
+/**
+ * Terms of use used to populate values of
+ * {@link FileTermsOfUse} in forms
+ * 
+ * @author madryk
+ */
+public class TermsOfUseForm implements Serializable {
+
+    private String typeWithLicenseId;
+    private String restrictType;
+    private String customRestrictText;
+    
+    // -------------------- GETTERS --------------------
+    
+    /**
+     * Returns terms of use type joined with license id
+     * (corresponds to joined value of {@link FileTermsOfUse#getTermsOfUseType()}
+     * and license id)<br/>
+     * For example string 'LICENSE_BASED:2' if corresponding
+     * {@link FileTermsOfUse} is license based and have assigned
+     * license with id 2)
+     */
+    public String getTypeWithLicenseId() {
```

Comment:
I understand that it is joined due to primeFace component?

Reply:
Yes. We have to somehow have this two informations inside single field